src/rulesets/LinearAlgebra/norm.jl

```diff
@@ -181,7 +182,11 @@ function rrule(
     return y, norm1_pullback
 end
 
-_norm1_back(x, y, Δy) = sign.(x) .* real(Δy)
+function _norm1_back(x, y, Δy)
+    ∂x = similar(x)
+    ∂x .= sign.(x) .* real(Δy)
+    return ∂x
+end
```

Comment:
It feels bad that we have to do this.
Should we open an issue on JuliaLang/julia about it?

Reply:
No, I think this is necessarily type-unstable, e.g.
```julia
julia> x = Diagonal([1.0, 2.0, 3.0]);

julia> x .* 0
3×3 Diagonal{Float64,Array{Float64,1}}:
 0.0   ⋅    ⋅
  ⋅   0.0   ⋅
  ⋅    ⋅   0.0

julia> x .* 10
3×3 Diagonal{Float64,Array{Float64,1}}:
 10.0    ⋅     ⋅
   ⋅   20.0    ⋅
   ⋅     ⋅   30.0

julia> x .* Inf
3×3 Array{Float64,2}:
  Inf  NaN   NaN
 NaN    Inf  NaN
 NaN   NaN    Inf

julia> x .* NaN
3×3 Array{Float64,2}:
 NaN  NaN  NaN
 NaN  NaN  NaN
 NaN  NaN  NaN
```

Reply:
Note that this breaks `gradient(norm, [1,2,3])`.